coalib/misc/Caching.py

```diff
@@ -0,0 +1,193 @@
+import calendar
+import os
+import pickle
+import time
+
+from coalib.misc.Constants import caching_db, changed_files_db
+from coalib.output.Tagging import get_tags_dir as get_pickle_dir
+
+
+def get_cache_data_path(log_printer, filename):
+    """
+    Get the full path of ``filename`` present in the user's cache directory.
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param filename:    The file whose path needs to be expanded.
+    :return:            Full path of the file, assuming it's present in the
+                        user's config directory.
+    """
+    return os.path.join(get_pickle_dir(log_printer), filename)
+
+
+def delete_cache_files(log_printer, files=[caching_db, changed_files_db]):
```

Comment:
I didn't know about immutable arguments before, but on thinking I think they can be used somewhere else: instead of writing to disk the list of changed files (`add_to_changed_files`), we can use immutable lists to keep updating the list. But yeah, shouldn't have used it over here.